pkg/rehearse/jobs.go

```diff
@@ -219,9 +225,116 @@ func ConfigureRehearsalJobs(toBeRehearsed config.Presubmits, ciopConfigs config.
 		}
 	}
 
+	if allowVolumes {
+		// for the remaining cluster types of each template, pick a `random` corresponding job to rehearse.
+		rehearsals = append(rehearsals, addRandomJobsForChangedTemplates(templates, masterConfigPresubmits, loggers, prNumber)...)
```

Comment:
Reading more into this, I do not understand why we are using `masterConfigPresubmits` here? That config is basically stale and can contain jobs that were removed or changed in the PR, there's no point in executing these... Why are we not using the PR version of the config?

Reply:
The job that we will pick up, is not supposed to be changed.